src/components/forms/AuthForm/AuthForm.tsx

```diff
@@ -17,20 +20,25 @@ const AuthForm = () => {
 
   const handleSignUpFormSumbit = async (signUpForm: SignUpSchemaType): Promise<void> => {
     const {confirmPassword, ...userData} = signUpForm;
-    const result = await dispatch(registerUser(userData)).unwrap();
-    if (!result.error) {
-      closeDialogAndGetUser();
+    try {
+      await dispatch(registerUser(userData)).unwrap();
+      closeDialogAndGetUser(SuccessToastMessages.REGISTER);
+    } catch (error) {
+      typeof error === 'string' && dispatch(openToast({message: error, type: ToastEnum.ERROR}));
     }
   };
 
   const handleSignInFormSubmit = async (signInForm: SignInSchemaType): Promise<void> => {
-    const result = await dispatch(loginUser(signInForm)).unwrap();
-    if (!result.error) {
-      closeDialogAndGetUser();
+    try {
+      await dispatch(loginUser(signInForm)).unwrap();
+      closeDialogAndGetUser(SuccessToastMessages.LOGIN);
+    } catch (error) {
+      typeof error === 'string' && dispatch(openToast({message: error, type: ToastEnum.ERROR}));
```

Comment:
`typeof error === 'string'` looks strange, you can`t do such checking everywhere.

Reply:
Ok, thanks. It is for TS only, but I will try to do smth another.

Reply:
Ok, we will discuss it today. `error as string` or `error.toString()`